framework/src/media/FileOutputDataSource.cpp

```diff
@@ -54,12 +54,14 @@ bool FileOutputDataSource::open()
 	return false;
 }
 
-void FileOutputDataSource::close()
+bool FileOutputDataSource::close()
 {
-	if (mFp) {
-		fclose(mFp);
+	if (mFp && fclose(mFp) == 0) {
```

Comment:
OK instead of 0, sometimes 0 means negative for another case.

Reply:
Instead of `fclose() == 0`, I modified it to `fclose() != EOF`